core/src/main/scala/io/finch/ContentTypeNegotiation.scala

```diff
@@ -0,0 +1,48 @@
+package io.finch
+
+import io.finch.internal.Accept
+import shapeless._
+
+trait ContentTypeNegotiation[ContentType] {
```

Comment:
Why does it have to be this order for typer parameters? Most of our components define content-type as a type-member vs a type-parameter here.

Reply:
It's the way to fix implicit divergence. Compiler seems to fail to resolve recursion of `mkCoproduct` because `A` parameter is stable from resolution to resolution.

Reply:
Interesting. How does this failure look like? An implicit not found, I'd guess. Yet I don't understand how is it different really. What if we change the order of params for `mkCoproduct`?

Reply:
Here is a minimal working example:

```scala
import shapeless.{:+:, CNil, Coproduct}

trait DivergenceTest[A] {

  type B

  def apply(): Unit

}

type Aux[a, b] = DivergenceTest[a] { type B = b }

implicit def mkCNil[A]: Aux[A, CNil] = new DivergenceTest[A] {
  type B = CNil

  def apply(): Unit = ()
}

implicit def mkCoproduct[A, H, T <: Coproduct]
  (implicit tail: Aux[A, T]): Aux[A, H :+: T] = {
  new DivergenceTest[A] {

    type B = H :+: T

    def apply(): Unit = tail()
  }
}

type C1 = Int :+: CNil
type C2 = String :+: Int :+: CNil
implicitly[Aux[String, C1]] //works
implicitly[Aux[String, C2]] //exception:  diverging implicit expansion for type Aux[String,C1] starting with method mkCNil
```

Reply:
I've replaced Aux pattern with two type parameters. It seems that there is no need for Aux anyway.